usr/lib/pkcs11/tpm_stdll/tpm_specific.c

```diff
@@ -3398,11 +3398,11 @@ CK_RV token_specific_get_mechanism_info(STDLL_TokData_t * tokdata,
 
 int token_specific_creatlock(void)
 {
-    CK_BYTE lockfile[PATH_MAX + (sizeof(LOCKDIR_PATH) - 1)
-                     + 2 * (sizeof(SUB_DIR) - 1)
-                     + (sizeof("///LCK..") - 1) + 1];
-    CK_BYTE lockdir[(sizeof(LOCKDIR_PATH) - 1) + (sizeof(SUB_DIR) - 1)
-                    + (sizeof("/") - 1) + 1];
+    char lockfile[PATH_MAX + (sizeof(LOCKDIR_PATH) - 1)
```

Comment:
PATH_MAX should be enough to store filenames. Maybe use snprintf some lines later in the code and check for truncation.

Reply:
LOCKDIR_PATH is provided by the build environment, technically you cant tell if it fits in PATH_MAX.